File: core/src/main/cpp/box2d/EdgeShape.cpp
#include "EdgeShape.h"
#include "Shape.h"
#include "Collision/Shapes/b2EdgeShape.h"

EdgeShape_M(jlong, newEdgeShape)(JNIEnv * , jclass) {
    return (jlong)(new b2EdgeShape());
}

EdgeShape_M(void, set) (JNIEnv
*env,
jobject obj, jfloat
x1,
jfloat y1, jfloat
x2,
jfloat y2
)
{
((b2EdgeShape*)env->
GetLongField(obj, shapePtr
))->
Set(b2Vec2(x1, y1), b2Vec2(x2, y2)
);
}

EdgeShape_M(void, jniGetVertex0) (JNIEnv
*env,
jobject obj, jfloatArray
vals)
{
jfloat *data = (jfloat * )
env->
GetPrimitiveArrayCritical(vals,
0);
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
data[0] = edge->m_vertex0.
x;
data[1] = edge->m_vertex0.
y;
env->
ReleasePrimitiveArrayCritical(vals, data,
0);
}
EdgeShape_M(void, setVertex0) (JNIEnv
*env,
jobject obj, jfloat
x,
jfloat y
)
{
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
edge->m_vertex0.
x = x;
edge->m_vertex0.
y = y;
}
EdgeShape_M(void, jniGetVertex1) (JNIEnv
*env,
jobject obj, jfloatArray
vals)
{
jfloat *data = (jfloat * )
env->
GetPrimitiveArrayCritical(vals,
0);
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
data[0] = edge->m_vertex1.
x;
data[1] = edge->m_vertex1.
y;
env->
ReleasePrimitiveArrayCritical(vals, data,
0);
}
EdgeShape_M(void, jniGetVertex2) (JNIEnv
*env,
jobject obj, jfloatArray
vals)
{
jfloat *data = (jfloat * )
env->
GetPrimitiveArrayCritical(vals,
0);
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
data[0] = edge->m_vertex2.
x;
data[1] = edge->m_vertex2.
y;
env->
ReleasePrimitiveArrayCritical(vals, data,
0);
}
EdgeShape_M(void, jniGetVertex3) (JNIEnv
*env,
jobject obj, jfloatArray
vals)
{
jfloat *data = (jfloat * )
env->
GetPrimitiveArrayCritical(vals,
0);
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
data[0] = edge->m_vertex3.
x;
data[1] = edge->m_vertex3.
y;
env->
ReleasePrimitiveArrayCritical(vals, data,
0);
}
EdgeShape_M(void, setVertex3) (JNIEnv
*env,
jobject obj, jfloat
x,
jfloat y
)
{
b2EdgeShape *edge = (b2EdgeShape *) env->GetLongField(obj, shapePtr);
edge->m_vertex3.
x = x;
edge->m_vertex3.
y = y;
}
EdgeShape_M(jboolean, hasVertex0)(JNIEnv * env, jobject
obj)
{
return ((b2EdgeShape*)env->
GetLongField(obj, shapePtr
))->
m_hasVertex0;
}
EdgeShape_M(void, setHasVertex0) (JNIEnv
*env,
jobject obj, jboolean
has)
{
((b2EdgeShape*)env->
GetLongField(obj, shapePtr
))->
m_hasVertex0 = has;
}
EdgeShape_M(jboolean, hasVertex3)(JNIEnv * env, jobject
obj)
{
return ((b2EdgeShape*)env->
GetLongField(obj, shapePtr
))->
m_hasVertex3;
}
EdgeShape_M(void, setHasVertex3) (JNIEnv
*env,
jobject obj, jboolean
has)
{
((b2EdgeShape*)env->
GetLongField(obj, shapePtr
))->
m_hasVertex3 = has;
}
	
	

File: core/src/main/java/com/ariasaproject/advancerofrpg/physics/box2d/EdgeShape.java
package com.ariasaproject.advancerofrpg.physics.box2d;

import com.ariasaproject.advancerofrpg.math.Vector2;

public class EdgeShape extends Shape {
    static final float[] vertex = new float[2];

    public EdgeShape() {
        addr = newEdgeShape();
    }

    EdgeShape(long addr) {
        this.addr = addr;
    }

    private native long newEdgeShape();

    public void set(Vector2 v1, Vector2 v2) {
        set(v1.x, v1.y, v2.x, v2.y);
    }

    public native void set(float v1X, float v1Y, float v2X, float v2Y);

    public void getVertex1(Vector2 vec) {
        jniGetVertex1(vertex);
        vec.x = vertex[0];
        vec.y = vertex[1];
    }

    private native void jniGetVertex1(float[] vertex);

    public void getVertex2(Vector2 vec) {
        jniGetVertex2(vertex);
        vec.x = vertex[0];
        vec.y = vertex[1];
    }

    private native void jniGetVertex2(float[] vertex);

    public void getVertex0(Vector2 vec) {
        jniGetVertex0(vertex);
        vec.x = vertex[0];
        vec.y = vertex[1];
    }

    private native void jniGetVertex0(float[] vertex);

    public void setVertex0(Vector2 vec) {
        setVertex0(vec.x, vec.y);
    }

    public native void setVertex0(float x, float y);

    public void getVertex3(Vector2 vec) {
        jniGetVertex3(vertex);
        vec.x = vertex[0];
        vec.y = vertex[1];
    }

    private native void jniGetVertex3(float[] vertex);

    public void setVertex3(Vector2 vec) {
        setVertex3(vec.x, vec.y);
    }

    public native void setVertex3(float x, float y);

    public native boolean hasVertex0();

    public native void setHasVertex0(boolean hasVertex0);

    public native boolean hasVertex3();

    public native void setHasVertex3(boolean hasVertex3);

    @Override
    public Type getType() {
        return Type.Edge;
    }

}
